codec_record.go

```diff
@@ -11,6 +11,9 @@ import (
 )
 
 func createDecoderOfRecord(cfg *frozenConfig, schema Schema, typ reflect2.Type) ValDecoder {
+	if dec := cfg.getDecoderFromCache(schema.CacheFingerprint(), typ.RType()); dec != nil {
```

Comment:
This will break if `DisableCaching` is set to true, and in fact as it is global, there is always an issue with disabled caching that is difficult to show in tests.

Further there is a race in parallel processing that will lead to SIGSEGV.